### DISCUSSION TITLE
PRI error when running in WinUI Desktop application

### DISCUSSION THREAD

**Discussion by KanniyappanP** [microsoft/WindowsAppSDK#1497] [Q&A] [answered]
Below PRI errors occurred when i try to build the WinUI application and my application has library project A with custom control libraries which referred as NuGet in library projects. 

Library project A -> Referred as Nuget B and Nuget C.   When i try to build the app with this nuget PRI errors occurred. 

Currently this can be resolved by repacking Nuget once again and resolved. 

Can you please tell why these errors are occurred and how to resolve? this?

`2>GENERATEPROJECTPRIFILE : error : PRI175: 0x80073b0f - Processing Resources failed with error: Duplicate Entry.
2>GENERATEPROJECTPRIFILE : error : PRI222: 0x80073b0f - Unspecified error occurred`

**Comment by andrewleader** [comment]
Hey @axelandrejs, can you look into this PRI build error?

**Comment by axelandrejs** [comment, marked as answer]
The error message is currently unhelpful. This is something we are planning to improve.

In terms of duplicated entry, this means that you have two resource candidates for the exact same language or scale. EG, if you have a resource ms-resource://foo, and you have two entries in an en-us resw file you'd get this error.

Sadly it is hard to diagnose this with the information given. If you can share a standalone repro project we can take a look. Otherwise, eliminating things until the error goes away is often the best way to track this down.

**Reply by IndianaGary** [reply]
I'm running into the same error on a Maui project. I have no idea what a PRI file is, or where to find it, or how or why it's generated. Could you provide a clue or some breadcrumbs for me to follow? I'd greatly appreciate it.

**Reply by axelandrejs** [reply]
What this means is that either you or more likely one of your dependencies brings in the same asset multiple times.

Assets are usually either images, or strings contained in .resw files. The files are usually marked by the language or scale they are for. Eg, en-us\foo.resw. I would look at the intermediate output, where the build process puts everything together and see if there is duplication. Eg, you can have the language encoded in the folder as I showed or in the filename itself.

Sadly this is mostly a manual exercise. There isn't really a good way to know which artifact is the duplicated one.

**Comment by IndianaGary** [comment]
Turns out that CommuniytToolkit.Maui was including Maui PRI data.

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
